lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Hunspell.java

```diff
@@ -417,13 +417,17 @@ private boolean checkCompoundRules(
 
   private boolean checkLastCompoundPart(
       char[] wordChars, int start, int length, List<IntsRef> words) {
-    IntsRef forms = dictionary.lookupWord(wordChars, start, length);
-    if (forms == null) return false;
+    IntsRef ref = new IntsRef(new int[1], 0, 1);
+    words.add(ref);
```

Comment:
reuse the "forms" object at the stack top, change its contents for each root candidate